### ISSUE TITLE
Make the role of "leap seconds" more obvious in DateTime and Instant

### ISSUE THREAD

**Issue by jonathanstowe** [Raku/doc#3881]
There was a question on Facebook where the poster was surprised that for a range of given date times the `DateTime.Instant.Int` varied from the perl `DateTime->epoch` by an increasing amount.  

Having looked at the documentation pages for both `DateTime` and `Instant` there doesn't appear to be anywhere where it is made explicit that leap seconds are in play at all, apart from the discussion of "Future Leap Seconds" in the `Instant` page.

It would probably be useful to have some introductory text in both `DateTime` and `Instant` that describes the role of the "leap seconds" and possibly also in the methods such as `.posix` which return the number of seconds describe why, due to the leap seconds, they will differ from `.Instance.Int`.

Haven't got any particular suggestions as to what the words should be.

**Comment by JJ** [comment]
Good idea.

**Comment by arkiuat** [comment]
I'm also planning a few short additions to DateTime.rakudoc, but the largest chunk of new text is planned for Instant.rakudoc, to be inserted under `=head1 Leap Seconds` immediately before the current `=head1 Future Leap Seconds` (which will change to `=head2`.  Here is the first draft, which I think is too long:

https://gist.github.com/arkiuat/f4baa44430034ffc512fb4414242e483

Soliciting suggestions for where/what to cut.

**Comment by arkiuat** [comment]
> My opinion would be that it is a fundamental misunderstanding of Instant
> for it to provide an Int method in the first place.  It's supposed to
> opaque outside of implementations.

This makes sense, but I'm not sure we can get that particular cat back into its particular bag. In the REPL, 

`.say for (my Instant $i).^methods.map(*.gist).elems`

prints 71, and the vast majority of these are methods that should never have been implemented by this standard, so it's not just `Int`.

**Comment by coke** [comment]
Someone please open a problem solving issue with that note, thanks.

**Comment by arkiuat** [comment]
@0rir, thank you for this excellent suggestion.  Three of your five points about the link's contents are specifically related to leap seconds, which only relate to UTC and not to any of the other time scales. Four really, because although unpredictability of future variance applies to UT1 as well, I think the interest here is focused on UTC.

So I'd like to add *two* entries to the glossary:

1. *time scale*, as you suggest, mentioning (and possibly linking to wikipedia for) UT1, TAI, maybe TT (fka ET), and POSIX time (which last is not really a timescale, strictly speaking).
2. *UTC*, which entry would contain all the discussion of leap- and deleted- seconds, their unpredictability, the advance notice that IERS gives before declaring one, and the expected dearth of leap seconds in near future.

**Comment by arkiuat** [comment]
In my aborted glossary PR, I did link to [http](https://en.wikipedia.org/wiki/Time_standard) (The "Time scale" link just gets a disambiguation page that has "Time standard" as its first link).

> I suggest you not go into Newton's or Einstein's theories; whether days or seconds are variable

I don't know why you mention Einstein, but you're probably right about Newton. My point there was to contrast 
1. UT1 which is the only timescale there was before the 20th c. and therefore traditional, but is incompatible with the idea of a fixed-length unit of time, with 
2. any of the fixed-length-unit timescales (of which there are way too many because of historical accidents, but contrasted to UT1 they're all the same: all use the SI second, just constant offsets from one another). 

The only way to make sense of leap seconds--which is the entire point of this issue--is to make clear that UTC is an attempt at compromise between these two incompatible systems, UT1 and fixed time. But you're right, calling it fixed time, or maybe SI-second time, would probably be better than alluding to Newton. (Side point, it's also the only way to make sense of the behavior of POSIX time around leap seconds, because POSIX time represents a different kind of attempt at the same compromise UTC tries to make.) 

> I suggest you not go into[...] whether days or seconds are variable

I believe the confusion about leap seconds is so persistent because we almost never point out that these are attempts at compromise, and neither do we point out the incompatibility that the compromise is supposed to address. 

It's not possible to adequately address this issue 3881 without pointing out that UT1 days are necessarily variable-length and SI seconds are necessarily not, so while I accept the rest of your guidance, I can't accept that one point.

**Comment by 0rir** [comment]
I agree with your _it's not possible_ point.   Accurately or not, I have gotten the idea that the terminology you use here is what you would have in the `doc` and that seems overly technical for the `doc`.  

The idea of _leap second_ is easily grasped by anyone understanding leap years.

The purely contextual information belonging in the `Instant` pages seems to be:
-  `Instant` implements the UTC time scale as a real number of seconds relative to a point in time.
-   The Gregorian calendar is used.
-  UTC is currently the commonest legal time on Earth. 
-   Seconds are of fixed size, but days vary...
-  ... so leap seconds are occasionally created
-  The IERS may add or delete a second to a  day.
-  The future is unsure.

All or most of that should be at the start of `Instant` and could be less than trice the size above.
Consequences of the above, should be demonstrated in code to promote clarity and conciseness.

TAI needs mention because it, in its raw form, is the underlying time scale.  Raw--ignoring 86400.
NTP time might be worth mentioning since that might inform those who had casual concern with
 NTP versus posix time when that was an issue.
IERS is the expert resource but they are externally pressured (this nit may be dismissed).

My underlying thoughts:
- Compared to popular languages, Raku-ing has little financial incentive, so growing the inner circle of gurus requires more rakoons becoming enamored;  so simple and internationized English are good.
- In 1987-8 the Times Square ball drop went "2, 1, Leap, 0", they have only done that once.
- My casual polling suggest that less than 30% of people have more than a heard-of-it awareness of leap seconds.

I don't claim to be an easy writer of English with or without the constraints of simplicity or international styling.

**Comment by librasteve** [comment]
I am fine with the proposed tweaks to my words.
